### DISCUSSION TITLE
Metrics stage for logs

### DISCUSSION THREAD

**Discussion by pmanno** [grafana-cold-storage/agent#1833] [Q&A]
Hi,

Promtail has a way to expose metrics from logs using the metrics stage.

https://grafana.com/docs/loki/latest/clients/promtail/stages/metrics/

I tried doing this, copying an existing promtail config and placing the metrics stage in with the logs being parsed.  This works perfectly in promtail, but I'm not seeing the metrics show up in mimir from the grafana-agent.  Is this implemented in grafana-agent or is there another integration I must enable to get the agent to scrape the custom metrics?

Thanks,
Paul

**Comment by tpaschalis** [comment]
Hey Paul! Sorry this fell through the cracks.

You'll need to define a new scrape job that to scrape those metrics; they're not automatically gathered, and the functionality is not exposed as an integration.
